be/src/vec/exec/volap_scanner.cpp

```diff
@@ -195,6 +193,10 @@ Status VOlapScanner::_init_tablet_reader_params(
               std::inserter(_tablet_reader_params.function_filters,
                             _tablet_reader_params.function_filters.begin()));
 
+    std::copy(_tablet->delete_predicates().cbegin(), _tablet->delete_predicates().cend(),
```

Comment:
Would this cause a wrong delete predicate working on a new-arrival rowset?

If we start a query with scanning rowset A (with a delete predicate), and just copy its delete predicate here, but a new rowset B is inserted meanwhile, would this delete predicate work on rowset B?

Reply:
The copied delete predicates only used for current scanner, in your case, current scanner can't see rowset
 B?